lib/house.rb

```diff
@@ -0,0 +1,27 @@
+class House
+  attr_reader :room,
+              :price,
+              :address,
+              :rooms
+
+
+  def initialize(price, address)
+    @price   = (price.delete("$").to_i)
+    @address = address
+    @rooms  = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    if price < 500000
+      false
+    end
+  end
+
+  def rooms_from_category(room)
+    :room.fetch(category)
+  end
```

Comment:
Anytime we need to work with a collection of things (`@rooms`) in this case, the first tool I turn to is iteration, either with `each` or another enumerable.  In this case, either of these solutions would be great:

```ruby
  def rooms_from_category(category)
    found_rooms = []
    @rooms.find_all do |room|
      if room.category == category
       found_rooms << room
      end
    end
    found_rooms
  end
```
OR
```ruby
  def rooms_from_category(category)
    @rooms.find_all do |room|
      room.category == category
    end
  end
```